odl/tomo/operators/spect_trafo.py

```diff
@@ -0,0 +1,318 @@
+except ImportError:
+    NIFTYREC_AVAILABLE = False
+
+import numpy as np
+
+from odl.operator import Operator
+from odl import uniform_discr_frompartition
+from odl.discr.lp_discr import DiscreteLp
+from odl.tomo.geometry.spect import ParallelHoleCollimatorGeometry
+
+
+__all__ = ('AttenuatedRayTransform', 'AttenuatedRayBackprojection',
+           'NIFTYREC_AVAILABLE')
+_SUPPORTED_IMPL = ('niftyrec', 'niftyrec_gpu')
+
+
+class AttenuatedRayTransform(Operator):
+
+    """The discrete attenuated Ray transform between L^p spaces.
+
+    The attenuated Ray transform is a variant of the classical
+    Ray transform  in which functions are integrated over lines
+    with respect to an exponential weight given an attenuation map.
+    """
+
+    def __init__(self, dom, geometry, attenuation=None,
+                 impl='niftyrec', **kwargs):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        dom : `DiscreteLp`
+            Domain of the forward projector, 3 dimensional.
+        geometry : `ParallelHoleCollimatorGeometry`
+            The geometry of the transform.
+        attenuation : domain `element-like`, optional
+            Linear attenuation map for SPECT.
+        impl : {'niftyrec', 'niftyrec_gpu'}, optional
+            Implementation back-end for the transform. Supported back-ends:
+
+            'niftyrec' : `NiftyRec`_ tomography toolbox using the `NiftyPy`_
+            Python wrapper and CPU
+
+            'niftyrec_gpu' : `NiftyRec`_ tomography toolbox using the
+            `NiftyPy`_ Python wrapper and GPU
+        psf : `array` of small size, optional
+            Point Spread Functions to be passed to NiftyRec
+            for the correction of the collimator-detector response
+
+        References
+        ----------
+        .._NiftyRec: http://niftyrec.scienceontheweb.net/wordpress/
+
+        .._NiftyPy: https://github.com/spedemon/niftypy
+
+        """
+        if not isinstance(dom, DiscreteLp):
+            raise TypeError('domain {!r} is not a `DiscreteLp`'
+                            ' instance.'.format(dom))
+
+        if not isinstance(geometry, ParallelHoleCollimatorGeometry):
+            raise TypeError('geometry {!r} is not a'
+                            '`ParallelHoleCollimatorGeometry` instance.'
+                            ''.format(geometry))
+
+        ran = uniform_discr_frompartition(geometry.partition)
+        super().__init__(dom, ran, linear=True)
+
+        impl, impl_in = str(impl).lower(), impl
+        if impl not in _SUPPORTED_IMPL:
+            raise ValueError('implementation {!r} not supported.'
+                             ''.format(impl_in))
+
+        if impl.startswith('niftyrec'):
+            if not NIFTYREC_AVAILABLE:
+                raise ValueError('NiftyRec back-end not available.')
+            if impl == 'niftyrec_gpu' and dom.shape != (128, 128, 128):
+                raise NotImplementedError('`nifryrec_gpu` back-end only '
```

Comment:
Use single ticks to show that niftyrec_gpu is a string. Also spelling error.

"'niftyrec_gpu' back-end only "